integration-tests/src/test/java/oracle/weblogic/kubernetes/ItServerStartPolicy.java

```diff
@@ -663,36 +735,175 @@ public void testStandaloneManagedRestartAlways() {
+    setupSample();
+    String wrongServerName = "ms1";
+    String regex = ".*" + wrongServerName + ".*\\s*is not part";
+
+    // verify that the script can not stop a non-exist server
+    String result =  assertDoesNotThrow(() ->
+        executeLifecycleScript(STOP_SERVER_SCRIPT, SERVER_LIFECYCLE, wrongServerName, "", false),
+        String.format("Failed to run %s", STOP_CLUSTER_SCRIPT));
+    assertTrue(verifyExecuteResult(result, regex),"The script shouldn't stop a server that doesn't exist");
+
+    // verify that the script can not start a non-exist server
+    result =  assertDoesNotThrow(() ->
+        executeLifecycleScript(START_SERVER_SCRIPT, SERVER_LIFECYCLE, wrongServerName, "", false),
+      String.format("Failed to run %s", START_SERVER_SCRIPT));
+    assertTrue(verifyExecuteResult(result, regex),"The script shouldn't start a server that doesn't exist");
+
+    // verify that the script can not stop a non-exist cluster
+    String wrongClusterName = "cluster-3";
+    regex = ".*" + wrongClusterName + ".*\\s*not part of domain";
+    result =  assertDoesNotThrow(() ->
+        executeLifecycleScript(STOP_CLUSTER_SCRIPT, CLUSTER_LIFECYCLE, wrongClusterName, "", false),
+        String.format("Failed to run %s", STOP_CLUSTER_SCRIPT));
+    assertTrue(result.contains("cluster cluster-3 is not part of domain"),
+        "The script shouldn't stop a cluster that doesn't exist");
+
+    // verify that the script can not start a non-exist cluster
+    result =  assertDoesNotThrow(() ->
+        executeLifecycleScript(START_CLUSTER_SCRIPT, CLUSTER_LIFECYCLE, wrongClusterName, "", false),
+        String.format("Failed to run %s", STOP_CLUSTER_SCRIPT));
+    assertTrue(result.contains("cluster cluster-3 is not part of domain"),
+        "The script shouldn't start a cluster that doesn't exist");
+
+    // verify that the script can not stop a non-exist domain
+    String domainName = "mii-start-policy" + "-123";
+    regex = ".*" + domainName + ".*\\s*not found";
+    result = assertDoesNotThrow(() ->
+        executeLifecycleScript(STOP_DOMAIN_SCRIPT, DOMAIN, null, "", false, domainName),
+        String.format("Failed to run %s", STOP_DOMAIN_SCRIPT));
+    assertTrue(verifyExecuteResult(result, regex),"The script shouldn't stop a domain that doesn't exist");
+
+    // verify that the script can not start a non-exist domain
+    result = assertDoesNotThrow(() ->
+        executeLifecycleScript(START_DOMAIN_SCRIPT, DOMAIN, null, "", false, domainName),
+        String.format("Failed to run %s", START_DOMAIN_SCRIPT));
+    assertTrue(verifyExecuteResult(result, regex),"The script shouldn't start a domain that doesn't exist");
+  }
+
+  /**
+   * Negative test to verify that the sample script can not start a server that exceeds the max cluster size
+   * e.g. the max cluster size is 2, the sample script can't start server 3
+   */
+  @Test
+  @DisplayName("Restart the configured cluster with serverStartPolicy")
```

Comment:
Modify the description for DisplayName()
Modify method name to testStartMSBeyondLimitNeg to testStartMangedServerBeyondMaxClusyterLimit()

Add a method for DynamicCluster size limit.

Reply:
fixed